python/lsst/summit/extras/slewTiming.py

```diff
@@ -0,0 +1,341 @@
+    'lsst.sal.ATMCS.command_startTracking',
+    'lsst.sal.ATMCS.command_stopTracking',
+    # 'lsst.sal.ATMCS.command_trackTarget',  # exclude the 20Hz data
+]
+
+
+def getMountPositionData(client, begin, end, prePadding=0, postPadding=0):
+    """Retrieve the mount position data from the EFD.
+
+    Parameters
+    ----------
+    client : `EfdClient`
+        The EFD client used to retrieve the data.
+    begin : `astropy.time.Time`
+        The start time of the data retrieval window.
+    end : `astropy.time.Time`
+        The end time of the data retrieval window.
+    prePadding : `float`, optional
+        The amount of time to pad before the begin time, in seconds.
+    postPadding : `float`, optional
+        The amount of time to pad after the end time, in seconds.
+
+    Returns
+    -------
+    alt, ax, rot : `tuple` of `pd.DataFrame`
+        A tuple containing the azimuth, elevation, and rotation data as
+        dataframes.
+    """
+    mountPosition = getEfdData(
+        client,
+        "lsst.sal.ATMCS.mount_AzEl_Encoders",
+        begin=begin,
+        end=end,
+        prePadding=prePadding,
+        postPadding=postPadding
+    )
+    nasmythPosition = getEfdData(
+        client, "lsst.sal.ATMCS.mount_Nasmyth_Encoders",
+        begin=begin,
+        end=end,
+        prePadding=prePadding,
+        postPadding=postPadding
+    )
+
+    az = mpts(mountPosition, 'azimuthCalculatedAngle', stride=1)
+    el = mpts(mountPosition, 'elevationCalculatedAngle', stride=1)
+    rot = mpts(nasmythPosition, 'nasmyth2CalculatedAngle', stride=1)
+    return az, el, rot
+
+
+def getCommandsIssued(client, commands, begin, end, prePadding, postPadding):
+    """Retrieve the commands issued within a specified time range.
+
+    Parameters
+    ----------
+    client : `EfdClient`
+        The client object used to retrieve EFD data.
+    commands : `list`
+        A list of commands to retrieve.
+    begin : `astropy.time.Time`
+        The start time of the time range.
+    end : `astropy.time.Time`
+        The end time of the time range.
+    prePadding : float
+        The amount of time to pad before the begin time.
+    postPadding : float
+        The amount of time to pad after the end time.
+
+    Returns
+    -------
+    commandTimes : `dict`
+        A dictionary where the keys are the timestamps of the commands and the
+        values are the corresponding commands.
+
+    Raises
+    ------
+    ValueError
+        Raise if there is already a command at a timestamp in the dictionary,
+        i.e. there is a collision.
+    """
+    commandTimes = {}
+    for command in commands:
+        data = getEfdData(
+            client,
+            command,
+            begin=begin,
+            end=end,
+            prePadding=prePadding,
+            postPadding=postPadding,
+            warn=False  # most commands will not be issue so we expect many empty queries
+        )
+        if not data.empty:
```

Comment:
Do you need this `if`?  Won't the `iterrows()` below just execute the body zero times if empty?  ("I think it is more clear this way" is an acceptable answer, since performance impact is clearly negligible.)

Reply:
No no, you're exactly right - I don't `pandas` all that often, and there was a place where I _did_ need to check, because I was merging dataframes, and if it was empty then the merge would fail due to the missing columns to merge on, and I'd just mis-extrapolated from that.